keras_cv/models/segmentation/segment_anything/sam_prompt_encoder.py

```diff
@@ -0,0 +1,321 @@
+# Copyright 2023 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import math
+
+from keras_cv.api_export import keras_cv_export
+from keras_cv.backend import keras
+from keras_cv.backend import ops
+from keras_cv.layers.serializable_sequential import SerializableSequential
+
+
+@keras_cv_export("keras_cv.layers.RandomFrequencyPositionalEmbeddings")
+class RandomFrequencyPositionalEmbeddings(keras.layers.Layer):
+    def __init__(self, num_positional_features, scale, **kwargs):
+        """Positional encoding using random spatial frequencies.
+
+        This layer maps coordinates/points in 2D space to positional
+        encodings using random spatial frequencies.
+
+        Args:
+            num_positional_features (int): Number of positional features
+                in the output.
+            scale (float): The standard deviation of the random frequencies.
+        """
+        super().__init__(**kwargs)
+        self.num_positional_features = num_positional_features
+        self.scale = scale
+        init_func = lambda *a, **kw: self.scale * ops.random.normal(
+            shape=(2, self.num_positional_features), dtype=self.dtype
+        )
+        self.positional_encoding_gaussian_matrix = self.add_weight(
+            name="positional_emcoding_gaussian_matrix",
+            shape=(2, self.num_positional_features),
+            dtype=self.dtype,
+            trainable=False,
+            initializer=init_func,
+        )
+
+        self.built = True
+
+    def __positional_encodings(self, coords):
```

Comment:
Maybe a layer?

Reply:
Is this useful elsewhere? Otherwise, it'd be easier to leave it as a private method.